lib/testing/fakeGpu.cpp

```diff
@@ -92,32 +94,46 @@ void FakeGpu::main_thread() {
     delta_ts.tv_sec = delta_ns / 1000000000;
     delta_ts.tv_nsec = delta_ns % 1000000000;
 
+    // Set up a random number generating for testing drops
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<> drop(0.0, 1.0);
+
     while (!stop_thread) {
-        int32_t* output = (int*)wait_for_empty_frame(out_buf, unique_name.c_str(), frame_id);
-        if (output == NULL)
-            break;
-
-        DEBUG("Simulating GPU buffer in {}[{}]", out_buf->buffer_name, frame_id);
-
-        allocate_new_metadata_object(out_buf, frame_id);
-        set_fpga_seq_num(out_buf, frame_id, fpga_seq);
-        set_stream_id_t(out_buf, frame_id, s);
-
-        // Set the two times
-        TIMESPEC_TO_TIMEVAL(&tv, &ts);
-        set_first_packet_recv_time(out_buf, frame_id, tv);
-        set_gps_time(out_buf, frame_id, ts);
-
-        // Fill the buffer with the specified mode
-        chimeMetadata* metadata = (chimeMetadata*)out_buf->metadata[frame_id]->metadata;
-        for (int freq_ind = 0; freq_ind < num_freq_in_frame; freq_ind++) {
-            gsl::span<int32_t> data(output + 2 * freq_ind * nprod_gpu,
-                                    output + 2 * (freq_ind + 1) * nprod_gpu);
-            pattern->fill(data, metadata, frame_count, freq + freq_ind);
-        }
 
-        // Mark full and move onto next frame...
-        mark_frame_full(out_buf, unique_name.c_str(), frame_id++);
+        // Test if this will be dropped or not, and only fill out the buffer
+        // (and advance) if it won't
+        if (drop(gen) < drop_probability) {
```

Comment:
I think @cubranic has created a stage that will randomly drop frames.  We might at some point want to switch to using that once it's merged into `develop` instead of having the test gen stages all have this functionality.  For now though I think this is ok to leave in.

Reply:
I did: `TestDropFrames` as part of #587 . It can drop specific frames or randomly with a given probability.

Reply:
Hmmmm. Okay. I might switch to using that at somepoint. Thanks!